sources/web/datalab/auth.ts

```diff
@@ -30,16 +31,35 @@ var oauth2Client: any = undefined;
 // These are the gcloud credentials and are not actually secret.
 let clientId = '32555940559.apps.googleusercontent.com';
 let clientSecret = 'ZmssLNjJy2998hD4CTg2ejr2';
-let gcloudDir = '/content/datalab/.config';
-let userCredFile = gcloudDir + '/credentials';
-let appCredFile = gcloudDir + '/application_default_credentials.json';
-let botoFile = '/etc/boto.cfg';
+
+// The application settings instance.
+var appSettings: common.Settings;
 
 let scopes = [
   'https://www.googleapis.com/auth/userinfo.email',
   'https://www.googleapis.com/auth/cloud-platform',
 ];
 
+// Path to the root of the gcloud configuration.
+function gcloudDir(): string {
+  return path.join(appSettings.datalabRoot, '/content/datalab/.config');
+}
+
+// Path for user-specific credentials.
+function userCredFile(): string {
```

Comment:
Does Datalab's auth flow works with this change? Here we save credential files to a custom dir but in pydatalab (https://github.com/googledatalab/pydatalab/blob/master/datalab/context/_utils.py#L53), we get credentials first from default credentials, and then try CLOUDSDK_CONFIG dir (which is not set unless we run setup-env.sh), and neither will retrieve credentials from the custom dir?

Or we actually require users to run gcloud auth login before running datalab outside docker, so that we never need to go through the auth flow?